src/create-sync.coffee

```diff
@@ -0,0 +1,94 @@
+    _.defaults(options || (options = {}), {
+      emulateHTTP: Backbone.emulateHTTP,
+      emulateJSON: Backbone.emulateJSON
+    })
+
+    # Default JSON-request options.
+    params = { type: type, dataType: 'json' }
+
+    # Ensure that we have a URL.
+    unless options.url
+      urlHost = hostOptions.host ? ''
+      url = _.result(model, 'url') || urlError()
+      params.url = urlHost + url
+
+    # Ensure that we have the appropriate request data.
+    if !options.data? && model && (method == 'create' || method == 'update' || method == 'patch')
+      params.contentType = 'application/json'
+      data = options.attrs || {}
+
+      if model.toServerJSON?
+        json = model.toServerJSON(method, options)
+      else
+        json = model.toJSON(options)
+
+      if model.paramRoot
+        data[model.paramRoot] = json
+      else
+        data = json
+
+      data.include = Utils.extractArray('include', options).join(',')
+      data.filters = Utils.extractArray('filters', options).join(',')
+      data.optional_fields = Utils.extractArray('optionalFields', options).join(',')
+
+      _.extend(data, options.params || {})
+
+      params.data = JSON.stringify(data)
+
+    # For older servers, emulate JSON by encoding the request into an HTML-form.
+    if options.emulateJSON
+      params.contentType = 'application/x-www-form-urlencoded'
+      params.data = if params.data then { model: params.data } else {}
+
+    # For older servers, emulate HTTP by mimicking the HTTP method with `_method`
+    # And an `X-HTTP-Method-Override` header.
+    if options.emulateHTTP && (type == 'PUT' || type == 'DELETE' || type == 'PATCH')
+      params.type = 'POST'
+      if options.emulateJSON
+        params.data._method = type
+      beforeSend = options.beforeSend
+      options.beforeSend = (xhr) ->
+        xhr.setRequestHeader 'X-HTTP-Method-Override', type
+        if beforeSend
+          beforeSend.apply this, arguments
+
+    # Clear out default data for DELETE requests, fixes a firefox issue where this
+    # exception is thrown: JavaScript component does not have a method named: “available”
+    if params.type == 'DELETE'
+      params.data = null
+
+    # Don't process data on a non-GET request.
+    if params.type != 'GET' && !options.emulateJSON
+      params.processData = false
+
+    # If we're sending a `PATCH` request, and we're in an old Internet Explorer
+    # that still has ActiveX enabled by default, override jQuery to use that
+    # for XHR instead. Remove this line when jQuery supports `PATCH` on IE8.
+    if params.type == 'PATCH' && window.ActiveXObject && !(window.external && window.external.msActiveXFilteringEnabled)
+      params.xhr = -> new ActiveXObject('Microsoft.XMLHTTP')
+
+    params.xhrFields = { withCredentials: hostOptions.withCredentials ? false }
+
+    # Make the request, allowing the user to override any Ajax options.
+    xhr = options.xhr = Backbone.ajax(_.extend(params, options))
```

Comment:
This method has always bothered me since it's an exact copy of  `Backbone.Sync`. One way to accomplish the modifications and cut down on duplication would be to add logic to `Backbone.ajax` and have it add in the extra data before the request is made. That way, less code would be duplicated and there's a focused area to debug when/if Backbone is updated.

Reply:
Could you clarify what you're suggesting? I'm not sure on what changes we would want to make to `Backbone.ajax`

Reply:
Ah, yeah that could be a good method to take

Reply:
@redjohn is it possible to override `Backbone.ajax` instead?

Reply:
I'm working on that today. Am I wrong in thinking that we could just override `Backbone.ajax` in the bigmaven repo and not even need to touch the other repos?

Reply:
Rewriting `.ajax` would replace https://github.com/mavenlink/brainstem-js/blob/cfd85d7d940d0ab2f80f757ece4c274d9d8054c8/src/sync.coffee so this repo would need to be modified with that approach.

Reply:
I'm able to get this working just by overriding `Backbone.sync` in the mavenlink repo: https://github.com/mavenlink/mavenlink/commit/8a03e0a8f223ae04ab3c684892df8ade15a7922f

I agree though that overriding `ajax` is preferable to overriding `sync`, especially if we weren't actually changing `sync` before. I think we could probably close this PR and open a new one for overriding `ajax` if we want to do that, but I don't think we need to pass any of the host info down into brainstem.

Reply:
https://github.com/mavenlink/brainstem-js/blob/master/src/sync.coffee can still leverage the work done in https://github.com/mavenlink/mavenlink/commit/8a03e0a8f223ae04ab3c684892df8ade15a7922f so there is still benefit in a PR (whether it's this one or another).